src/java/org/apache/cassandra/index/sai/disk/vector/VectorsView.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.vector;
+
+public interface VectorsView extends AutoCloseable
```

Comment:
I'd rather not call it a View since that has a specific meaning in JVector that is related but distinct.  VectorSupplier maybe?

Reply:
Done. I'm curious, what is the relevant point of distinction?

Reply:
the JVector View is primarily about the state of the graph and only incidentally about providing a method to fetch vectors